lib/util/host-rules.ts

```diff
@@ -155,7 +155,6 @@ export function find(search: HostRuleSearch): HostRuleSearchResult {
     .forEach((rule) => {
       res = merge(res, rule);
     });
-  delete res.hostType;
```

Comment:
this can break other code, because the returned object is directly used.

@rarkins any better solution?

Reply:
Yes, I suspected that... Would there be any other way of getting the hostType, if configured, for a given host? Otherwise we can't tell gitlab and github enterprise apart, as both can have completely custom domains, and I think that doing a network request and checking for the result is not a good solution at all...

It would be possible to define a `findHostRule` method which returns the `HostRule`, and `find` is just a wrapper around `findHostRule` which casts to `HostRuleSearchResult`? But then we have another method exported...

Reply:
actually there is `hostType`, I'll adjust the MR

Reply:
This should be much better, as it also does not take any assumptions about how the repository is accessed (which was implicitly done by checking for the `token`)